apps/room/views/mixins/room_membership.py

```
@@ -0,0 +1,20 @@
+from django.contrib.auth.mixins import AccessMixin
+from django.http import HttpResponseForbidden
+
+
+class RoomMembershipRequiredMixin(AccessMixin):
+    """Ensure the requesting user belongs to the room (or is a superuser)."""
+
+    def dispatch(self, request, *args, **kwargs):
```

Ruff (0.14.14)

[warning] Missing type annotation for `*args`

(ANN002)